src/bp/ui-studio/src/web/components/ChatEmulator/Message.jsx

```diff
@@ -17,24 +17,29 @@ export default class Message extends React.Component {
     return <span className={classnames(style.message, style.other)}>{message.type} (can't render)</span>
   }
 
-  renderIntentLink(intent) {
-    if (intent === 'N/A') {
-      return intent
+  renderFinalDecision(finalDecision) {
+    if (!finalDecision) {
+      return 'N/A'
     }
+    const { source, sourceDetails, decision } = finalDecision
 
     const QNA_PREFIX = '__qna__'
-    const link = intent.includes(QNA_PREFIX)
-      ? `/modules/qna#search:${intent.replace(QNA_PREFIX, '')}`
-      : `/modules/nlu/Intents#search:${intent}`
+    const link =
```

Comment:
I know you didn't really change this but IMO it's a bit weird that the view logic of the chat emulator is aware of the QNA module because a module

Reply:
Quick workaround, best way would be to let module "register" sources & lookup paths, in the future.